lib/perl/Genome/Model/Tools/BioSamtools/AlignmentSummaryV2.pm

```diff
@@ -577,8 +577,13 @@ sub _prepare_split_sorted_bed_files {
         my $chrom = $1;
 
         unless ($open_handles{$chrom}) {
+            if(scalar keys %open_handles >= 100) {
+                $_->close foreach values %open_handles;
```

Comment:
Maybe it should just cache a filehandle for consecutive writes for the same chromosome but close and open anytime the chromosome changes?  I don't know what the data looks like.